src/gui/widget_paint.c

```diff
@@ -122,7 +122,7 @@ static LCUI_BOOL Widget_IsPaintable(LCUI_Widget w)
 	    s->shadow.spread > 0) {
 		return TRUE;
 	}
-	return w->proto && w->proto->paint;
+	return w->proto && w->proto->paint && w->proto->name;
```

Comment:
The condition is that the widget has a custom paint method, so just check to see if the widget's paint method is equal to the default paint method.

```diff
+ static LCUI_WidgetPrototype default_proto = NULL;
+
+ if (!default_proto) {
+         default_proto = LCUIWidget_GetPrototype(NULL);
+ }
- return w->proto && w->proto->paint && w->proto->name;
+ return w->proto->paint != default_proto->paint;
```

Reply:
The `default_proto` variable can be moved into the `self` structure.

```diff
static struct LCUI_WidgetRenderModule {
	LCUI_BOOL active;
+ 	LCUI_WidgetPrototype default_proto;
	RBTree groups;
	LinkedList rects;
} self = { 0 };
```

So, we don't have to check whether default_proto is NULL here,

```diff
void LCUIWidget_InitRenderer(void)
{
	RBTree_Init(&self.groups);
	RBTree_OnCompare(&self.groups, OnCompareGroup);
	RBTree_OnDestroy(&self.groups, OnDestroyGroup);
	LinkedList_Init(&self.rects);
+	self.default_proto = LCUIWidget_GetPrototype(NULL);
	self.active = TRUE;
}
```

Reply:
return w->proto->paint && proto->paint != default_proto->paint;

maybe we should check for correct proto->paint handler yet, in future it would be availability be creates abstract prototypes without paint hander